### DISCUSSION TITLE
MONITORINFOEX

### DISCUSSION THREAD

**Discussion by msedi** [microsoft/CsWin32#695] [Q&A] [answered]
I know that there already is a question around GetMonitorInfo, see #188. The problem here is that the GetMonitorInfo accepts two structures, the MONITORINFO and MONITORINFO ex. From the documented method signature:

```
BOOL GetMonitorInfoA(
  [in]  HMONITOR      hMonitor,
  [out] LPMONITORINFO lpmi
);
```

this is not evident, it is only internally distinguished by inspecting the `cbSize` field.#188  
I'm not into the WinMetaData, but can this be seen somewhere in the metadata and if yes, would it be possible to generate two invoke methods with different MONITORINFO + EX parameters?

**Comment by AArnott** [comment, marked as answer]
The metadata does not and likely will not ever indicate when a win32 method that takes a pointer to a struct may also accept a pointer to a different struct unless the header files also indicate that (which at least in this case doesn't happen AFAIK). 
For now, you can always create your `MONITORINFOEX` struct in C# then pass a pointer to it into the `GetMonitorInfo` method that takes the pointer. I don't know yet what we can do to make this easier or more discoverable.

**Reply by msedi** [reply]
Thanks for pointing this out.
